src/axom/sidre/core/ItemCollection.hpp

```diff
@@ -125,14 +126,23 @@ namespace sidre
  * \class ItemCollection
  *
  * \brief ItemCollection is an abstract base class template for holding
- *        a collection of items of template parameter type TYPE.  Derived
+ *        a collection of items of template parameter type T.  Derived
  *        child classes can determine how to specifically store the items.
  *
  *************************************************************************
  */
-template <typename TYPE>
+template <typename T>
 class ItemCollection
 {
+public:
+  using value_type = T;
+
+  // Forward declare iterator classes and helpers
+  class iterator;
+  class const_iterator;
+  class iterator_adaptor;
```

Comment:
Note: I used `adaptor` vs. `adapter` for consistency w/ the C++ std library.

See: https://stackoverflow.com/questions/2076321/adapter-or-adaptor